llama-index-core/llama_index/core/base/llms/types.py

```diff
@@ -26,6 +30,47 @@ class MessageRole(str, Enum):
 
 
 # ===== Generic Model Input - Chat =====
+class TextBlock(BaseModel):
+    type: str = "text"
+
+    content: str
+
+
+class ImageBlock(BaseModel):
+    type: str = "image"
+
+    image: Optional[str] = None
+    image_path: Optional[str] = None
+    image_url: Optional[str] = None
+    image_mimetype: Optional[str] = None
+
+    @property
+    def content(self) -> str:
+        """Get the image represented as a bytes string, path, or URL."""
+        if self.image is not None:
+            return self.image
+        elif self.image_path is not None:
+            return self.image_path
+        elif self.image_url is not None:
+            # load image from URL
+            return self.image_url
+        else:
+            raise ValueError("No image found in the chat message!")
+
+    def resolve_image(self) -> ImageType:
+        """Resolve an image such that PIL can read it."""
+        if self.image is not None:
+            return BytesIO(base64.b64decode(self.image))
```

Comment:
I feel like there was a serialization reason that we keep the image as a string, but I'm not totally remembering why

Reply:
That's completely fine, but I would add a proper getter/setter API so we are guaranteed the underlying image storage is always serializable.

Reply:
since the typing for image, image_str, and image_path is all string, we are guaranteed that its always serializable (although I would not promote storing serialized byte strings of images of course)